src/vstest.console/InProcessVsTestConsoleWrapper.cs

```diff
@@ -0,0 +1,1101 @@
+    { }
+
+    internal InProcessVsTestConsoleWrapper(
+        ConsoleParameters consoleParameters,
+        ITranslationLayerRequestSender requestSender,
+        ITestRequestManager? testRequestManager,
+        Executor executor,
+        ITestPlatformEventSource testPlatformEventSource)
+    {
+        EqtTrace.Info("VsTestConsoleWrapper.StartSession: Starting VsTestConsoleWrapper session.");
+
+        _testPlatformEventSource = testPlatformEventSource;
+        _testPlatformEventSource.TranslationLayerInitializeStart();
+
+        // Start communication.
+        _requestSender = requestSender;
+        var port = _requestSender.InitializeCommunication();
+        if (port <= 0)
+        {
+            // Close the sender as it failed to host server.
+            _requestSender.Close();
+            throw new TransationLayerException("Error hosting communication channel.");
+        }
+
+        // Fill the parameters.
+        consoleParameters.ParentProcessId = Process.GetCurrentProcess().Id;
+        consoleParameters.PortNumber = port;
+
+        // Start vstest.console.
+        // TODO: under VS we use consoleParameters.InheritEnvironmentVariables, we take that
+        // into account when starting a testhost, or clean up in the service host, and use the
+        // desired set, so all children can inherit it.
+        foreach (var pair in consoleParameters.EnvironmentVariables)
+        {
+            Environment.SetEnvironmentVariable(pair.Key, pair.Value);
+        }
+
+        string someExistingFile = typeof(InProcessVsTestConsoleWrapper).Assembly.Location;
+        var args = new VsTestConsoleProcessManager(someExistingFile).BuildArguments(consoleParameters);
+        // Skip vstest.console path, we are already running in process, so it would just end up
+        // being understood as test dll to run. (it is present even though we don't provide
+        // dotnet path, because it is a .dll file).
+        args = args.Skip(1).ToArray();
+
+        // We standup the client, and it will allocate port as normal that we will never use.
+        // This is just to avoid "duplicating" all the setup logic that is done in argument
+        // processors before client processor. It created design mode client, and stores it as
+        // single instance.
+        // We connect back to the client but never use that connection, it only serves as
+        // "await" to make sure the design mode client is already initialized.
+        Task.Run<int>(() => executor.Execute(args));
+        WaitForConnection();
+
+        // Set the test request manager here.
+        TestRequestManager = testRequestManager;
+        if (TestRequestManager == null)
+        {
+            TPDebug.Assert(
+                (DesignModeClient?)DesignModeClient.Instance != null,
+                "DesignModeClient.Instance is null");
+            TestRequestManager = ((DesignModeClient)DesignModeClient.Instance).TestRequestManager;
```

Comment:
Could we make instance being directly of type `DesignModeClient`? It seems a little weird to have it returning the interface when we know it's always of type 'DesignModeClient`.

Reply:
Fixed.

Reply:
Should be fixed now.